core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/keystore/persistence/KeyStoreRegistryPersistenceManager.java

```diff
@@ -0,0 +1,421 @@
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try {
+            return registry.resourceExists(path);
+        } catch (RegistryException e) {
+            throw new SecurityException("Error checking the existence of key store: " + keyStoreName, e);
+        }
+    }
+
+    /**
+     * Get the password for the given key store name.
+     *
+     * @param keyStoreName Key store name.
+     * @return KeyStore Password.
+     * @throws SecurityException If there is an error while getting the key store password.
+     */
+    public char[] getKeyStorePassword(String keyStoreName, int tenantId) throws SecurityException {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        try {
+            if (isKeyStoreExistsInRegistry(keyStoreName, registry)) {
+                String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+                Resource resource = registry.get(path);
+                return getKeyStorePasswordFromRegistryResource(resource, keyStoreName);
+            } else {
+                throw new SecurityException("Key Store with a name: " + keyStoreName + " does not exist.");
+            }
+        } catch (RegistryException e) {
+            throw new SecurityException("Error getting password for key store: " + keyStoreName, e);
+        }
+    }
+
+    /**
+     * Get the private key password as a character array for the given key store name.
+     *
+     * @param keyStoreName Key store name.
+     * @param tenantId     Tenant Id.
+     * @return Private key password as a character array.
+     * @throws SecurityException If an error occurs while getting the private key password.
+     */
+    public char[] getPrivateKeyPassword(String keyStoreName, int tenantId) throws SecurityException {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try {
+            Resource resource = registry.get(path);
+            String encryptedPassword = resource.getProperty(RegistryResources.SecurityManagement.PROP_PRIVATE_KEY_PASS);
+            return decryptPassword(encryptedPassword);
+        } catch (RegistryException e) {
+            throw new SecurityException("Error getting private key password for key store: " + keyStoreName, e);
+        }
+    }
+
+    private char[] getKeyStorePasswordFromRegistryResource(Resource resource, String keyStoreName) {
+
+        String encryptedPassword = resource.getProperty(RegistryResources.SecurityManagement.PROP_PASSWORD);
+        if (encryptedPassword == null) {
+            throw new SecurityException("Key Store Password of " + keyStoreName + " does not exist.");
+        }
+        return decryptPassword(encryptedPassword);
+    }
+
```

Comment:
Move `decryptPassword()` impl to over here.

Reply:
Addressed with https://github.com/wso2/carbon-kernel/compare/63f3d33d1ed8704022f55645be6b047a005641fb..6c4a7e5fc5efca69a114e5d3a08289b8dc809be7